### PR TITLE
[facebook] Update TS and auto-generate docs

### LINKED ISSUE #13606: [docs] [expo-facebook] logEventAsync
### Summary

Hi team, i notice that expo-facebook recently added logEventAsync() & logPurchaseAsync(). I tried to use those function however there was nothing trigger to FB Event Manager. Also there wasn't any documentation about it too. May i know the feature still in development? 

I also notice there is also getAdvertiserIDAsync()and i tried to use the Id to trigger Facebook Activity Event. However it also fail to capture at FB Event Manger too.

Below are the version that i used.
expo SDK: 42
expo-facebook: 11.3.1

I hope that we able to use those method to sending app activities back to facebook.

Thanks.

### Link to the related docs page

https://docs.expo.io/versions/latest/sdk/facebook/

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by cruzach**
# Why

closes https://github.com/expo/expo/issues/13606

# How

updated TS, regenerated docs

# Test Plan

`yarn dev`

### CODE REVIEW

**Review by tsapeta — APPROVED** [review]
👍

File: docs/pages/versions/v42.0.0/sdk/facebook.md [hunk 1]
```diff
@@ -174,34 +85,14 @@ async function logIn() {
...
-}
+```js
+import * as Facebook from 'expo-facebook';
 ```
 
+<APISection packageName="expo-facebook" />
+
 ## Types
 
 ### FacebookAuthenticationCredential
```
Comment (Simek) [inline docs/pages/versions/v42.0.0/sdk/facebook.md:98]: @cruzach It looks like `FacebookAuthenticationCredential` type is duplicated on the page. `APISection` outputs this type too:
- https://docs.expo.io/versions/latest/sdk/facebook/#facebookauthenticationcredential
- https://docs.expo.io/versions/latest/sdk/facebook/#facebookauthenticationcredential-1

Comment (cruzach) [reply docs/pages/versions/v42.0.0/sdk/facebook.md:98]: thanks @Simek ! fixed here- https://github.com/expo/expo/commit/260aff86930bcf48b6037dfcea5b988f5acfe761

**Comment by Simek** [comment]
Also it looks like `FacebookInitializationOptions` type is not exported, and it's missing on the generated page:
* https://docs.expo.io/versions/latest/sdk/facebook/#initializeasyncoptionsorappid-appname

**Comment by cruzach** [comment]
thanks again @Simek 🙏 fixed on `master`

**Comment by Simek** [comment]
@cruzach Unfortunately there are still types missing:

Screenshot 2021-07-16 201754

You can export them directly, or add the [`Facebook.types.ts`](https://github.com/expo/expo/blob/master/packages/expo-facebook/src/Facebook.types.ts) file to the source files list in `GenerateDocsAPIData.ts` (so the first argument becomes an array). You can see the example in this changeset, before the refactor:
* https://github.com/expo/expo/commit/1ef472c36504020442ffa364bd2071a6345936ee#diff-2d291bc209aa01d0c07db93c42cac3a446687c5e8b4c6f12be05e6a7dac33f70L92

**Pull request merged**